server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRedirect.java

```diff
@@ -153,41 +155,61 @@
 
 public abstract class AbstractRedirect extends GenericMailet {
 
+    private static final char LINE_BREAK = '\n';
+
     protected abstract boolean isNotifyMailet();
 
-    /**
-     * Gets the expected init parameters.
-     *
-     * @return null meaning no check
-     */
-    protected String[] getAllowedInitParameters() {
-        return null;
-    }
+    protected abstract String[] getAllowedInitParameters();
 
-    /**
-     * Controls certain log messages.
-     */
     protected boolean isDebug = false;
 
-    /**
-     * Holds the value of the <code>static</code> init parameter.
-     */
     protected boolean isStatic = false;
 
+    private static enum SpecialAddressKind {
```

Comment:
Not really the responsability of this enum, it's more logic to have it in AddressMarker.